astropy/convolution/utils.py

```diff
@@ -262,11 +263,13 @@ def discretize_oversample_2D(model, x_range, y_range, factor=10):
     Discretize model by taking the average on an oversampled grid.
     """
     # Evaluate model on oversampled grid
-    x = np.arange(x_range[0] - 0.5 * (1 - 1 / factor),
-                  x_range[1] + 0.5 * (1 + 1 / factor), 1. / factor)
+    x = np.linspace(x_range[0] - 0.5 * (1 - 1 / factor),
+                    x_range[1] + 0.5 * (1 - 1 / factor),
+                    num=(x_range[1] - x_range[0] + 1) * factor)
+    y = np.linspace(y_range[0] - 0.5 * (1 - 1 / factor),
+                    y_range[1] + 0.5 * (1 - 1 / factor),
+                    num=(y_range[1] - y_range[0] + 1) * factor)    
```

Comment:
If factor=1 and for x_range from say 0 to 3, the old method returned [0, 1, 2] but the new method returns [0, 1, 2, 3] so this doesn't seem quite right? Can we add a test to make sure that discretize_oversample with factor=1 is the same as discretize_center?

Reply:
For this case (factor=1), both old and new methods will give [0, 1, 2, 3]. The last element will be eventually popped out (as it was done before), and the return will give the pixel values centered at [0, 1, 2]. This is the same as discertize_center()